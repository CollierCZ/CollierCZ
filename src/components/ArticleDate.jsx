import React from "react"
import { css } from '@emotion/core'
import moment from "moment"

class ArticleDate extends React.Component {
  render() {
    const { date, prefix } = this.props;
    return (
      <span>
        {prefix}
        <time
          css={css`
            display: inline-block;
            white-space: nowrap;
            font-size: inherit;
            padding: 0;
            margin: 0;
            border: none;
            @media only screen and (max-width: 500px) {
              display: none;
            }
          `}
          dateTime={moment(new Date(date)).format("YYYY-MM-DD")}
        >
          {moment(new Date(date)).format("DD MMMM YYYY")}
        </time>
      </span>
    );
  }
}

export default ArticleDate;
