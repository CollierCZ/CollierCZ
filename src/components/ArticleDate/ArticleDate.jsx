import React from "react";
import moment from "moment";
import "./ArticleDate.css";

class ArticleDate extends React.Component {
  render() {
    const { date } = this.props;
    return (
      <time
        className="article-date"
        dateTime={moment(new Date(date)).format("YYYY-MM-DD")}
      >
        {moment(new Date(date)).format("DD MMMM YYYY")}
      </time>
    );
  }
}

export default ArticleDate;
