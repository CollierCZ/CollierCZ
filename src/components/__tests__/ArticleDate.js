import React from "react"
import { shallow } from "enzyme"
import ArticleDate from "../ArticleDate"

describe("ArticleDate", () => {
  it("renders correctly", () => {
    const component = shallow (<ArticleDate date="2019-06-03T00:00:00Z" prefix="Published " />)
    expect(component).toMatchSnapshot()
  })
})