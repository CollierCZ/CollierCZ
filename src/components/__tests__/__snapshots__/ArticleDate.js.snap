// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDate renders correctly 1`] = `
<span>
  Published 
  <time
    css={
      Array [
        "
          display: inline-block;
          white-space: nowrap;
          font-size: inherit;
          padding: 0;
          margin: 0;
          border: none;
          @media only screen and (max-width: 500px) {
            display: none;
          }
        ",
      ]
    }
    dateTime="2019-06-03"
  >
    03 June 2019
  </time>
</span>
`;
